src/TrackingCollectionTests/TrackingCollectionTests.cs

```diff
@@ -1797,25 +1798,43 @@ public void Removing()
 
         expectedCount = 13;
         col.RemoveItem(GetThing(5));
-        evt.WaitOne();
+        Assert.True(evt.WaitOne(40));
         evt.Reset();
         CollectionAssert.AreEqual(col, new List<Thing> {
             GetThing(4, 4),
             GetThing(6, 6),
             GetThing(9, 9),
         });
 
-        col.Filter = null;
+        col.Filter = (item, position, list) => true;
 
         expectedCount = 14;
+        col.RemoveItem(GetThing(0));
+        Assert.True(evt.WaitOne(40));
+        evt.Reset();
+
+        Assert.AreEqual(8, col.Count);
+        CollectionAssert.AreEqual(col, new List<Thing> {
+            GetThing(1, 1),
+            GetThing(3, 3),
+            GetThing(4, 4),
+            GetThing(6, 6),
+            GetThing(7, 7),
+            GetThing(8, 8),
+            GetThing(9, 9),
+            GetThing(10, 10),
+        });
+
+        col.Filter = null;
+
+        expectedCount = 15;
```

Comment:
This unit test is getting rather long - might it be best to split this test out into a separate test method?